docs/migrations/v3-to-v4.md

```diff
@@ -25,6 +25,25 @@ This is an API breaking change and as such IBC application developers will have
 
 ## IBC Light Clients
 
+### Light client and ClientState
+
 The `VerifyUpgradeAndUpdateState` function has been modified. The client state and consensus state return value has been removed. 
 
 Light clients **must** set the updated client state and consensus state in the client store after verifying a valid client upgrade.
+
+The `CheckHeaderAndUpdateState` function has been split into 4 new functions: `VerifyClientMessage`, `CheckForMisbehaviour`, `UpdateState`, 
+`UpdateStateOnMisbehaviour`
+
+Light client implementations now need to manage setting of client and consensus states for these interface functions `UpdateState`, `UpdateStateOnMisbehaviour`, `VerifyUpgradeAndUpdateState`, `CheckSubstituteAndUpdateState`
+
```

Comment:
```suggestion
```

I think we can remove this and state the light client must handle all management of client and consensus states, including setting updated client/consensus states